presto-iceberg/src/test/java/com/facebook/presto/iceberg/procedure/TestFastForwardBranchProcedure.java

```diff
@@ -0,0 +1,278 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg.procedure;
+
+import com.facebook.presto.iceberg.IcebergConfig;
+import com.facebook.presto.iceberg.IcebergQueryRunner;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.google.common.collect.ImmutableMap;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.facebook.presto.iceberg.CatalogType.HADOOP;
+import static com.facebook.presto.iceberg.IcebergQueryRunner.getIcebergDataDirectoryPath;
+import static java.lang.String.format;
+
+public class TestFastForwardBranchProcedure
+        extends AbstractTestQueryFramework
+{
+    public static final String ICEBERG_CATALOG = "test_hadoop";
+    public static final String TEST_SCHEMA = "tpch";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return IcebergQueryRunner.createIcebergQueryRunner(ImmutableMap.of(), HADOOP, ImmutableMap.of());
+    }
+
+    public void createTable(String tableName)
+    {
+        assertUpdate("CREATE TABLE IF NOT EXISTS " + tableName + " (id integer, value VARCHAR)");
+    }
+
+    public void dropTable(String tableName)
+    {
+        assertQuerySucceeds("DROP TABLE IF EXISTS " + TEST_SCHEMA + "." + tableName);
+    }
+
+    @Test
+    public void testFastForwardBranchUsingPositionalArgs()
+    {
+        String tableName = "fast_forward_table_test";
+        createTable(tableName);
+        try {
+            assertUpdate("INSERT INTO " + tableName + " VALUES (1, 'a')", 1);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (2, 'b')", 1);
+
+            Table table = loadTable(tableName);
+            table.refresh();
+
+            table.manageSnapshots().createBranch("testBranch").commit();
+            assertUpdate("INSERT INTO " + tableName + " VALUES (3, 'c')", 1);
+            table.refresh();
+
+            assertQuery("SELECT * FROM " + tableName + " ORDER BY id", " VALUES (1, 'a'), (2, 'b'), (3, 'c')");
+            assertQuery("SELECT * FROM " + tableName + " FOR SYSTEM_VERSION AS OF 'testBranch'" + " ORDER BY id", " VALUES (1, 'a'), (2, 'b')");
+
+            String fromBranch = "testBranch";
+            String toBranch = "main";
+            assertUpdate(format("CALL system.fast_forward('%s', '%s', '%s', '%s')", TEST_SCHEMA, tableName, fromBranch, toBranch));
+
+            // now testBranch branch should have 3 entries same as main
+            assertQuery("SELECT * FROM " + tableName + " FOR SYSTEM_VERSION AS OF 'testBranch'" + " ORDER BY id", " VALUES (1, 'a'), (2, 'b'), (3, 'c')");
```

Comment:
final `+` not needed?
```suggestion
            assertQuery("SELECT * FROM " + tableName + " FOR SYSTEM_VERSION AS OF 'testBranch' ORDER BY id", " VALUES (1, 'a'), (2, 'b'), (3, 'c')");
```